src/UtilityFunctions.cpp

```diff
@@ -41,6 +41,12 @@ int sumGeneratedNumbers(std::unique_ptr<NumberGenerator> gen, int num)
 
 int sumNumbersFromGenerators(std::vector<GeneratorType> const & types, int num)
 {
+    double sum = 0;
+    for(GeneratorType type: types)
```

Comment:
You can use `const GeneratorType & type` or `const auto & type` to avoid additional copy.